Update build intructions for mixedrefs by default, remove ARM sections

pshipton opened the pull request:
Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

pshipton commented:
@keithc-ca @sharon-wang pls review before I copy this to all the places that duplicate it.

keithc-ca commented:
I don't think the mixed-references paragraph reads very well. I suggest the tag should be about object reference representation and then go on to describe the choices and mention the defaults.

pshipton commented:
@keithc-ca pls take another look.

sharon-wang reviewed (approved):
This reads well to me.

Although not part of the build instructions, would it be helpful to link to the object reference mode command line options https://www.eclipse.org/openj9/docs/xcompressedrefs/ that can be used with a mixed references build?

pshipton commented:
I've added a docs link and other changes. Keith suggested using 'non-compressed references' or 'uncompressed references', I've changed it to use 'non-'.

keithc-ca commented on doc/build-instructions/Build_Instructions_V8.md:
```diff
@@ -154,9 +154,13 @@ bash configure --with-freemarker-jar=/root/freemarker.jar --with-boot-jdk=/usr/l
 :warning: The path in the example --with-boot-jdk= option is appropriate for the Docker installation. If not using the Docker environment, set the path appropriate for your setup, such as "/<my_home_dir>/bootjdk8" as setup in the previous instructions.
 
-:pencil: **Non-compressed references support:** If you require a heap size greater than 57GB, enable a noncompressedrefs build with the `--with-noncompressedrefs` option during this step.
+:pencil: **Mixed and compressed references support:** A 64-bit build can support mixed references which includes both compressed and non-compressed references, compressed references only, or non-compressed references only. When using mixed references the [compressed state](https://www.eclipse.org/openj9/docs/gc_overview/#compressed-references) is selected at runtime. Mixed references is the defau ...
```
I think it would be clearer to invert the comment about using cmake:

> Note that support for mixed references is not compatible with `--with-cmake=no`.

pshipton replied:
Updated.

pshipton replied:
I'm thinking to re-write that as: Note that `--with-cmake=no` cannot be used to build mixed references.

pshipton replied:
Pushed that change.

keithc-ca replied:
I think I figured out how this sentence is *supposed* to be parsed:

> A 64-bit build can support mixed references which includes both compressed and non-compressed references, compressed references only, or non-compressed references only.

Unfortunately, it's not clear that compressed-only and non-compressed-only are separate from mixed support. Perhaps it would help to reorder the clauses as they're listed below

> A 64-bit build can be created to support:
> * compressed references (only)
> * non-compressed references (only)
> * both compressed and non-compressed references (selected as the VM is created)

pshipton replied:
Pushed a change.

pshipton replied:
and tweaked it.

keithc-ca replied:
Much improved: thanks.

pshipton commented:
I've made some additional changes related to freemarker.jar. Once this looks good I'll propagate these changes to other platforms and versions.

keithc-ca commented:
I like where you're headed: carry on. ;-)

pshipton commented:
Pls have another look, all platforms and versions are updated.

keithc-ca commented on doc/build-instructions/Build_Instructions_V16.md:
````diff
@@ -609,4 +650,153 @@ JCL      - d0f36677cbe based on jdk-16+35)
+
+Start a Docker container from the **openj9aarch64** image with the following command, where `<host_directory>` is the directory that contains `openj9-openjdk-jdk16` in your local system:
+```
+docker run -v /<host_directory>/openj9-openjdk-jdk16:/root/openj9-openjdk-jdk16 -it openj9aarch64
````
Please remove the slash before `<host_directory>` for consistency with build instructions for other versions.

keithc-ca commented on doc/build-instructions/Build_Instructions_V16.md:
```diff
@@ -609,4 +650,153 @@ JCL      - d0f36677cbe based on jdk-16+35)
+               --with-freetype-lib=${OPENJ9_CC_DIR}/${OPENJ9_CC_PREFIX}/libc/usr/lib \
+               --with-boot-jdk=/root/bootjdk11 \
+               --with-build-jdk=/root/bootjdk11 \
+               --with-cmake=no \
```
Using `--with-cmake=no` requires `--with-freemarker-jar`.

keithc-ca commented on doc/build-instructions/Build_Instructions_V8.md:
````diff
@@ -432,28 +445,36 @@ When you have all the source files that you need, run the configure script, whic
 ```
 bash configure --disable-ccache \
                --with-boot-jdk=/cygdrive/c/<path_to_jdk8> \
-               --with-freemarker-jar=/cygdrive/c/temp/freemarker.jar \
````
We're getting a little ahead of ourselves here: cmake is not yet supported for win32: Should this change wait?

pshipton replied:
Just a little ahead I hope, the cmake support for win32 seems just about ready to merge. I've restored the freemarker line for now. Also pushed changes to address all the other comments.

keithc-ca commented on doc/build-instructions/Build_Instructions_V11.md:
````diff
@@ -522,6 +544,14 @@ The following dependencies can be installed by using [Homebrew](https://brew.sh/
+
+```
+wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tgz
+gtar -xzf freemarker.tgz freemarker-2.3.8/lib/freemarker.jar --strip-components=2
````
It appears that GNU tar accepts both options, but the documentation seems to favour `--strip-components`, so we should use that consistently.

pshipton commented on doc/build-instructions/Build_Instructions_V11.md:
````diff
@@ -109,7 +108,7 @@ version used in the build.
-3. Download and setup **freemarker.jar** into a directory.
+3. Only when building with `--with-cmake=no`, download and setup `freemarker.jar` into a directory.
 ```
 cd /<my_home_dir>
````
Pushed another change, hopefully I've got them all now.

keithc-ca replied:
There are still a number of places that could be fixed, e.g.
* `--with-freemarker-jar=/<path_to>/freemarker.jar`
* `--with-boot-jdk=/<path_to_boot_JDK>`
* `docker run -v /<host_directory>/openj9-openjdk-jdk11:/root/openj9-openjdk-jdk11`
* `export JAVA_HOME=/<path_to_build_JDK>`

Search for `/<` to find them all.

pshipton replied:
ok, fixed them all and pushed.
